### ISSUE TITLE
SNS .util.SignatureChecker fails when it shouldn't due to bad parsing

### ISSUE THREAD

**Issue by acinader** [aws/aws-sdk-java#510]
I'm creating an SNS message in the the AWS SNS ui.  My client is attempting to validate the raw json using  ```com.amazonaws.services.sns.util.SignatureChecker.verifyMessageSignature(String message, PublicKey publicKey)``` method.

It's failing with the following: 

```
java.lang.RuntimeException: Cannot process message of type String
	at com.amazonaws.services.sns.util.SignatureChecker.verifySignature(SignatureChecker.java:105)
```

The problem is that ```private Map<String, String> parseJSON(String jsonmessage)``` finds more than one ```fieldName``` of ```Type``` and it overwrites the correct Type, "Notification", with the incorrect Type, "String", that is found while parsing the ```MessageAttributes``` section.

Below is a sample message that will trip the bug.

I'd suggest fixing by having a POJO for a message that jackson maps into.  Not sure what the pull request policy is, but I can cook up if it isn't a waste of time.

the message 

```
{
  "Type" : "Notification",
  "MessageId" : "85c8859f-4201-5b71-ab61-cbed06183f8e",
  "TopicArn" : "arn:aws:sns:us-east-1:810605503585:just_testing",
  "Subject" : "another test",
  "Message" : "this is a test",
  "Timestamp" : "2015-09-15T16:05:45.300Z",
  "SignatureVersion" : "1",
  "Signature" : "UH6scc3l4MjKtfJ+0gcrEW+/IxMfJ05KG9G9wCKIyEkTV+W5tLBNrsp9uvnqHxX2OGCaVFqEJgYwmHa5O65tyUvtRJRUbo/fFSZKI3GHQUWd8RZ4ro0NYyE/DQTkBufKG406mrmZRndS9ajckLRgFF4gbL7bbHas+GGdONsSHylvV2fuOUuiTnlQk87N1hDTsQa56Mic/eFF4BEylZ5tqr5qNvEDctR+2BkWlL+fUwfU6PSfiJBJFIlYABr+tSLHmk0xqkg5Ue7mlmoyvFPLMMf/SKoSJo8uLXrkie2pT0b/eIV3RfauvD7vwYwZhUTDOQSxXovV3u/P2ZA0z/T0Sg==",
  "SigningCertURL" : "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-bb750dd426d95ee9390147a5624348ee.pem",
  "UnsubscribeURL" : "https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:810605503585:just_testing:6521e74c-2dea-4732-99b8-071c12356149",
  "MessageAttributes" : {
    "AWS.SNS.MOBILE.MPNS.Type" : {"Type":"String","Value":"token"},
    "AWS.SNS.MOBILE.MPNS.NotificationClass" : {"Type":"String","Value":"realtime"},
    "AWS.SNS.MOBILE.WNS.Type" : {"Type":"String","Value":"wns/badge"}
  }
}
```

**Comment by acinader** [comment]
#513 solves this particular problem.  I imagine that if there could be a message body that'll still mess up the parse of the json object, but I couldn't figure out how to make one using the console ui.  In any event, this should be an improvement.

**Comment by acinader** [comment]
resolved by #513
